arrow-libs/integrations/arrow-raise-ktor-server/src/commonMain/kotlin/arrow/raise/ktor/server/request/RaisingParameterProvider.kt

```diff
@@ -0,0 +1,75 @@
+package arrow.raise.ktor.server.request
+
+import arrow.core.raise.Raise
+import io.ktor.http.*
+import io.ktor.util.reflect.*
+import kotlin.jvm.JvmInline
+import kotlin.jvm.JvmName
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+public abstract class RaisingParameterProvider internal constructor(
```

Comment:
🤔 at the moment this only raises when a delegated property is used.
Do we want to raise as soon as a delegate is acquired? 

This could be achieved by the same process as the accumulating delegate provider but returning an already-resolved delegate.